core/src/utils/animation/animation-utils.ts

```diff
@@ -1,9 +1,25 @@
+export const getAnimationPrefix = (el: HTMLElement): string => {
```

Comment:
I think we can replace the implementation with the following:

```
const getAnimationPrefix = (): string => {
  const el = document.documentElement;
  const supportsUnprefixed = (el.style as any).animationName !== undefined;
  const supportsWebkitPrefix = (el.style as any).webkitAnimationName !== undefined;
  
  return (!supportsUnprefixed && supportsWebkitPrefix) ? '-webkit-' : '';
};

const animationPrefix: string = getAnimationPrefix();
```

This prevents us from a) having to pass in an element and b) having to check for a prefix every time we call `setStyleProperty` or `removeStyleProperty`.

Reply:
I disagree with your suggestion. With this const variable is not possible to use ssr. In the first version of this pr I used an cache value. But at the end it's just a comparison to undefined. this is no expensive operation or not a more expensive operation than test for a cache value.

Reply:
Ionic Animations wasn't compatible with SSR to begin with (see: https://github.com/ionic-team/ionic/blob/master/core/src/utils/animation/animation.ts#L67-L68). It's something on the list to address, but I think it's fine for now.

Reply:
I'd just rather not constantly recheck the same value when we know the vendor prefix isn't going to change mid-session.

Reply:
What if we went back to the old way of passing in `el` to `getAnimationPrefix`, but then we cache the vendor prefix value and return it if it's already defined? That way we aren't doing additional unnecessary checks and we don't stray further from SSR support.